### PR TITLE
Bug/unit tests

### PR DESCRIPTION
**Pull request by payneBrandon**
# PR Details
## Description
This PR addresses failing and non-included unit tests. When running Maven commands ('mvn test' for instance), the existing code base runs successfully. However, due to configuration a large number of unit tests were being exempt. This PR updates that configuration to pull in all unit tests, and fixes issues found within those unit tests causing failures.

## Related Issue
#459 

## Motivation and Context
This change is required to successfully run all unit tests.

## How Has This Been Tested?
Tested locally on multiple development environments. Also ran in remote development environment as defined by code base.

## Types of changes

- [X] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
[ODE Contributing Guide](https://github.com/usdot-jpo-ode/jpo-ode/blob/bugfix/Pull_request_template/docs/contributing_guide.md) 
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Review by TonyEnglish — APPROVED** [review]
Thank you

**Comment by payneBrandon** [comment]
I've added a bug fix for #458 into this PR as well as it is a small fix. The additional code is under OdeSvcsApplication.java, line 52-53 and simply registers missing security protocols from the previously updated SNMP library. Note that this fix was tested locally and in WYDOT with success.

**Review by TonyEnglish — APPROVED** [review]
Thanks for fixing the SNMP issue as well!

File: jpo-ode-common/src/main/java/us/dot/its/jpo/ode/util/JsonUtils.java [hunk 1]
```diff
@@ -45,32 +49,45 @@ public JsonUtilsException(String string, Exception e) {
...
+      mapper_noNulls.setSerializationInclusion(Include.NON_NULL);
    }
 
    public static String toJson(Object o, boolean verbose) {
-
       // convert java object to JSON format,
       // and returned as JSON formatted string
-      return verbose ? gsonVerbose.toJson(o) : gsonCompact.toJson(o);
+      try {
+         return verbose ? mapper.writeValueAsString(o) : mapper_noNulls.writeValueAsString(o);
```
Comment (dan-du-car) [inline jpo-ode-common/src/main/java/us/dot/its/jpo/ode/util/JsonUtils.java:75]: Not sure why we need this update. It does the same thing as the existing code.

Comment (payneBrandon) [reply jpo-ode-common/src/main/java/us/dot/its/jpo/ode/util/JsonUtils.java:75]: This swaps over to using the Jackson library. There was an odd mixture using both the Gson and Jackson libraries, and they did not always mix. When I got the unit tests to run, large numbers of them were failing and many were due to this mismatch of serializations. You are correct that the new code performs the same as previous in this instance and that was intentional - this update ensures compatibility and chooses a single serializer (Jackson) over the current duality.

File: jpo-ode-svcs/src/test/java/us/dot/its/jpo/ode/traveler/TimTransmogrifierTest.java [hunk 2]
```diff
@@ -206,15 +217,13 @@ public void testConvertToXMLASD(@Capturing DateTimeUtils capturingDateTimeUtils,
...
...ngs><encodings><elementName>Ieee1609Dot2Data</elementName><elementType>Ieee1609Dot2Data</elementType><encodingRule>COER</encodingRule></encodings><encodings><elementName>AdvisorySituationData</elementName><elementType>AdvisorySituationData</elementType><encodingRule>UPER</encodingRule></encodings></encodings></metadata><payload><dataType>us.dot.its.jpo.ode.plugin.j2735.DdsAdvisorySituationData</dataType><data><AdvisorySituationData><dialogID>156</dialogID><seqID>5</seqID><groupID>00000000</groupID><requestID>7876BA7F</requestID><recordID>00000000</recordID><timeToLive>1</timeToLive><serviceRegion/><asdmDetails><asdmID>7876BA7F</asdmID><asdmType>2</asdmType><distType>03</distType><startTime><year>0</year><month>0</month><day>0</day><hour>0</hour><minute>0</minute></startTime><stopTime><year>0</year><month>0</month><day>0</day><hour>0</hour><minute>0</minute></stopTime><advisoryMessage><Ieee1609Dot2Data><protocolVersion>3</protocolVersion><content><unsecuredData><MessageFrame><messageId>31</messageId><value><TravelerInformation/></value></MessageFrame></unsecuredData></content></Ieee1609Dot2Data></advisoryMessage></asdmDetails></AdvisorySituationData></data></payload></OdeAsn1Data>";
```
Comment (dan-du-car) [inline jpo-ode-svcs/src/test/java/us/dot/its/jpo/ode/traveler/TimTransmogrifierTest.java:225]: Where is this additional <ttl>thirtyminutes</ttl> from metadata from?

Comment (payneBrandon) [reply jpo-ode-svcs/src/test/java/us/dot/its/jpo/ode/traveler/TimTransmogrifierTest.java:225]: This test creates a new SDW object on line 201, and sets that in the ServiceRequest. When serialized, the SDW object is using it's get methods to inject properties. The SDW object has a default value returned when calling the getTTL() method:

image

File: jpo-ode-common/src/main/java/us/dot/its/jpo/ode/util/JsonUtils.java (hunk 1 above)
Comment (dan-du-car) [reply jpo-ode-common/src/main/java/us/dot/its/jpo/ode/util/JsonUtils.java:75]: Thanks!

**Pull request merged**